pkg/index/mysql/mysql.go

```diff
@@ -0,0 +1,106 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/index"
+	_ "github.com/lib/pq"
+)
+
+func New(cfg *config.MySQL) (index.Indexer, error) {
+	dataSource := getMySQLSource(cfg)
+	db, err := sql.Open("mysql", dataSource)
+	if err != nil {
+		return nil, err
+	}
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+	_, err = db.Exec(mysqlTable)
+	if err != nil {
+		return nil, err
+	}
+	return &indexer{db}, nil
+}
+
+const mysqlTable = `
+	CREATE TABLE IF NOT EXISTS indexes(
+	id INT
+		AUTO_INCREMENT
+		PRIMARY KEY
+		COMMENT 'Unique identifier for a module line',
+
+	module VARCHAR(255)
+		NOT NULL
+		COMMENT 'Name of the module',
+
+	version VARCHAR(255)
+		NOT NULL
+		COMMENT 'Name of the module',
+
+	timestamp TIMESTAMP(6)
+		COMMENT 'Date and time when the module was first created',
+
+	INDEX (timestamp),
+	UNIQUE INDEX module_version (module, version)
+	) CHARACTER SET utf8;
+`
+
+type indexer struct {
+	db *sql.DB
+}
+
+func (i *indexer) Index(ctx context.Context, mod, ver string) error {
+	const op errors.Op = "sql.Index"
```

Comment:
should the op mention the type of db, rather than just `sql`?

Reply:
It should probably read `mysql.Index`, because the pattern has mostly been `<pkg>.<method>`